app/premium/_components/ServersList.tsx

```diff
@@ -0,0 +1,47 @@
+"use server";
+
+import { DiscordGuild } from "../_types";
+import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar";
+import { SelectItem } from "@/components/ui/select";
+import { getServer } from "@/helpers/cache/redis";
+import Image from "next/image";
+import { useEffect, useState } from "react";
+
+export async function ServersList() {
+  const [serversData, setServersData] = useState<DiscordGuild[]>([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const servers: any = await getServer();
+      console.log(servers, "servers");
+      setServersData(servers);
+    };
+
+    fetchData();
+  }, []);
```

Comment:
Server components can't use hooks.
Fetch the data as the component is being rendered.
```suggestion
      const serversData = await getServer();
```